inference-mistral.py

```diff
@@ -27,13 +28,24 @@ def __init__(self, model_path: str, checkpoint_path: str) -> None:
         self.ft_model = PeftModel.from_pretrained(model, checkpoint_path).eval()
 
     def generate_response(self, prompt: str, max_tokens: int = 50) -> str:
-        inputs = self.tokenizer(prompt, return_tensors='pt')
+        inputs = self.prepare_prompt(prompt)
         with torch.no_grad():
             logger.info('Generating up to %d tokens...', max_tokens)
             outputs = self.ft_model.generate(**inputs, max_length=max_tokens, pad_token_id=2)
         logger.info('Decoding...')
         return self.tokenizer.decode(outputs[0], skip_special_tokens=True)
 
+    def prepare_prompt(self, prompt):
```

Comment:
```suggestion
    def prepare_prompt(self, prompt: str):
```

I'm not sure what the return type is since we're splatting it with `**` to `self.ft_model.generate`...

Reply:
It returns something like `transformers.tokenization_utils_base.BatchEncoding`. I prefer not to specify the type here